### ISSUE TITLE
Dovecot + prefixes

### ISSUE THREAD

**Issue by jyaworski** [nextcloud/mail#185]
See https://github.com/owncloudarchive/mail/issues/1468 for the other issue I filed.

In the [README](https://github.com/nextcloud/mail#dovecot-imap):

```text
If your Dovecot IMAP server prefixes all folders with INBOX, Nextcloud Mail does not work correctly.
```

Why is this, and can this behavior be fixed?

**Comment by Henni** [comment]
@ChristophWurst I'm having the same issue. Any idea? Can we fix this to increase usability?

**Comment by ChristophWurst** [comment]
> Any idea?

Nope.

**Comment by fti7** [comment]
Any updates here? I cant change the Prefix Option on the Mailserver :-(

**Comment by ChristophWurst** [comment]
Hey @fti7,

you're lucky, I'm currently re-writing the code that has been holding us back from resolving this limitation: https://github.com/nextcloud/mail/pull/271. Once that PR has been merged, it should be easy to support other folder structures with prefixes too 🚀

**Comment by oparoz** [comment]
Fixed via #271

**Comment by reubenlillie** [comment]
Please pardon my confusion, @ChristophWurst and @oparoz, but has the Dovecot namespace prefixing issue been fixed via #271 or not? That is, does the "not working correctly" that I'm experiencing relate to what #271 attempted to fix, or is my issue, like #372, a separate kind of bug?

In any case, I'm having difficulty similar to #372 on my instance (unless I'm also misunderstanding #372):

### Expected behaviour
"Show all folders" displays folders beyond inbox (e.g., Archive, Sent, Drafts, Junk, etc.)

### Actual behaviour
"Show all folders" displays no more folders, instead only "Collapse folders" toggle prompt.

### Mail app

**Mail app version:** 0.6.4

**Mailserver or service:** Dovecot

**Transport security - IMAP:** SSL 

**Transport security - SMTP:** SSL

**Number of accounts:** 1 (testing for development, ~250 accounts planned for client)

### Server configuration
### Server configuration
**Operating system**: Linux server.***{DOMAIN}*** 2.6.32-696.el6.x86_64 #1 SMP Tue Mar 21 19:29:05 UTC 2017 x86_64

**Web server:** Apache (cgi-fcgi)

**Database:** mysql 5.6.35

**PHP version:** 7.1.5
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, readline, Reflection, SPL, session, standard, cgi-fcgi, bcmath, calendar, ctype, curl, dom, mbstring, ftp, gd, gettext, imap, json, exif, mcrypt, mysqlnd, PDO, Phar, posix, pspell, SimpleXML, soap, sockets, sqlite3, tidy, tokenizer, xml, xmlwriter, xsl, zip, mysqli, pdo_mysql, pdo_sqlite, wddx, xmlreader, xmlrpc

**Nextcloud version:** 12.0.0 - 12.0.0.29

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:** https://nextcloud.com/install/

**Signing status:**
<details>
	<summary>Signing status</summary>

```
[]
```

</details>

**List of activated apps:**
<details>
	<summary>App list</summary>
```
Enabled:
 - activity: 2.5.2
 - bruteforcesettings: 1.0.2
 - calendar: 1.5.3
 - comments: 1.2.0
 - contacts: 1.5.3
 - dav: 1.3.0
 - external: 2.0.3
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_pdfviewer: 1.1.1
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - issuetemplate: 0.2.1
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - mail: 0.6.4
 - nextcloud_announcements: 1.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - provisioning_api: 1.2.0
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0

Disabled:
 - admin_audit
 - encryption
 - files_external
 - user_external
 - user_ldap
```
</details>

**The content of config/config.php:**
<details>
	<summary>Config report</summary>

```
{
    "instanceid": "ocn3td528q7q",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud.***{DOMAIN}***"
    ],
    "datadirectory": "\/home\/***{DOMAIN}***\/public_html\/cloud\/data",
    "overwrite.cli.url": "https:\/\/cloud.***{DOMAIN}***",
    "dbtype": "mysql",
    "version": "12.0.0.29",
    "dbname": "***{DOMAIN}***_nextcloud",
    "dbhost": "localhost",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "mail_domain": "***{DOMAIN}***",
    "mail_smtpmode": "php",
    "mail_smtpauthtype": "LOGIN",
    "mail_from_address": "cloud",
    "app.mail.imaplog.enabled": true,
    "app.mail.smtplog.enabled": true,
    "app.mail.imap.timeout": 20,
    "app.mail.smtp.timeout": 2,
    "app.mail.transport": "php-mail"
}
```
</details>

**Are you using external storage, if yes which one:** files_external is disabled

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

### Client configuration
**Browser:** Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0

**Operating system:**

### Logs
#### Web server error log (data/horde_imap.log)
<details>
	<summary>Web server error log</summary>
	```
	------------------------------
>> Wed, 31 May 2017 19:05:16 +0000
>> Connection to: imap://mail.***{DOMAIN}***.org:993/
>> Server connection took 0.0262 seconds.
S: * OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE NAMESPACE AUTH=PLAIN AUTH=LOGIN] Dovecot ready.
C: 1 AUTHENTICATE PLAIN [INITIAL CLIENT RESPONSE (username: cloud@pghnaz.org)]
S: 1 OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE SORT SORT=DISPLAY THREAD=REFERENCES THREAD=REFS THREAD=ORDEREDSUBJECT MULTIAPPEND URL-PARTIAL CATENATE UNSELECT CHILDREN NAMESPACE UIDPLUS LIST-EXTENDED I18NLEVEL=1 CONDSTORE QRESYNC ESEARCH ESORT SEARCHRES WITHIN CONTEXT=SEARCH LIST-STATUS BINARY MOVE NAMESPACE NOTIFY SPECIAL-USE COMPRESS=DEFLATE QUOTA] Logged in
>> Command 1 took 0.0632 seconds.
C: 2 LIST () "" (*) RETURN (SPECIAL-USE)
S: * LIST () "." INBOX
S: * LIST (\UnMarked \Archive) "." INBOX.Archive
S: * LIST (\UnMarked \Junk) "." INBOX.spam
S: * LIST (\UnMarked \Junk) "." INBOX.Junk
S: * LIST (\UnMarked \Trash) "." INBOX.Trash
S: * LIST (\UnMarked \Drafts) "." INBOX.Drafts
S: * LIST (\UnMarked \Sent) "." INBOX.Sent
S: 2 OK List completed (0.001 + 0.000 + 0.001 secs).
>> Command 2 took 0.0018 seconds.
C: 3 STATUS INBOX.Junk (MESSAGES)
S: * STATUS INBOX.Junk (MESSAGES 0)
S: 3 OK Status completed (0.001 + 0.000 secs).
>> Command 3 took 0.0005 seconds.
C: 4 STATUS INBOX.Junk (MESSAGES)
S: * STATUS INBOX.Junk (MESSAGES 0)
S: 4 OK Status completed (0.001 + 0.000 secs).
>> Command 4 took 0.0003 seconds.
C: 5 STATUS INBOX.spam (MESSAGES)
S: * STATUS INBOX.spam (MESSAGES 0)
S: 5 OK Status completed (0.001 + 0.000 secs).
>> Command 5 took 0.0003 seconds.
C: 6 STATUS INBOX (MESSAGES RECENT UIDNEXT UIDVALIDITY UNSEEN)
S: * STATUS INBOX (MESSAGES 2 RECENT 0 UIDNEXT 3 UIDVALIDITY 1496251615 UNSEEN 0)
S: 6 OK Status completed (0.001 + 0.000 secs).
>> Command 6 took 0.0008 seconds.
C: 7 LOGOUT
S: * BYE Logging out
S: 7 OK Logout completed (0.001 + 0.000 secs).
>> Command 7 took 0.0009 seconds.
	```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
	<summary>Nextcloud log</summary>
	```
	empty
	```
</details>

#### Browser log
<details>
	<summary>Browser log</summary>
	```
	Insert your browser log here, this could for example include:

	a) The javascript console log
Content Security Policy: Directive ‘frame-src’ has been deprecated. Please use directive ‘child-src’ instead.  (unknown)
JQMIGRATE: Migrate is installed, version 1.4.0  core.js:7:542
Starting Mail …  mail.min.js:157:26485
initializing cache…  mail.min.js:152:21641
[ui] "content:loading" Array [  ]  mail.min.js:148:14867
[ui] "notification:request" Array [  ]  mail.min.js:148:14867
[account] "entities" Array [  ]  mail.min.js:148:14867
[ui] "sidebar:loading" Array [  ]  mail.min.js:148:14867
[ui] "search:set" Array [ "" ]  mail.min.js:148:14867
[ui] "content:loading" Array [ "Loading Inbox" ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[message] "entities" Array [ Object, Object, Object ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[ui] "folder:changed" Array [  ]  mail.min.js:148:14867
[ui] "navigation:newmessage:show" Array [  ]  mail.min.js:148:14867
[ui] "sidebar:accounts" Array [  ]  mail.min.js:148:14867
[ui] "foldercontent:show" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ null ]  mail.min.js:148:14867
[message] "bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[message] "fetch:bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[ui] "message:loading" Array [  ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ Object ]  mail.min.js:148:14867
[message] "entity" Array [ Object, Object, 2 ]  mail.min.js:148:14867
[ui] "composer:events:undelegate" Array [  ]  mail.min.js:148:14867
[message] "flag" Array [ Object, Object, Object, "unseen", false ]  mail.min.js:148:14867
[ui] "messagesview:messageflag:set" Array [ 2, "unseen", false ]  mail.min.js:148:14867
[ui] "message:show" Array [ Object ]  mail.min.js:148:14867
[message] "load" Array [ Object, Object, Object ]  mail.min.js:148:14867
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead.  mail
[keyboard] "keyup" Array [ Object, 9 ]  mail.min.js:148:14867
[ui] "search:set" Array [ "" ]  mail.min.js:148:14867
[ui] "content:loading" Array [ "Loading Inbox" ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[message] "entities" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "foldercontent:show" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ null ]  mail.min.js:148:14867
[message] "bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[message] "fetch:bodies" Array [ Object, Object, Array[2] ]  mail.min.js:148:14867
[ui] "composer:leave" Array [  ]  mail.min.js:148:14867
[ui] "message:loading" Array [  ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ Object ]  mail.min.js:148:14867
[message] "entity" Array [ Object, Object, 2 ]  mail.min.js:148:14867
[ui] "composer:events:undelegate" Array [  ]  mail.min.js:148:14867
[message] "flag" Array [ Object, Object, Object, "unseen", false ]  mail.min.js:148:14867
[ui] "messagesview:messageflag:set" Array [ 2, "unseen", false ]  mail.min.js:148:14867
[ui] "message:show" Array [ Object ]  mail.min.js:148:14867
[message] "load" Array [ Object, Object, Object ]  mail.min.js:148:14867
[message] "entities" Array [ Object, Object, Object ]  mail.min.js:148:14867
[ui] "messagesview:messages:update" Array [  ]  mail.min.js:148:14867
[folder] "setactive" Array [ Object, Object ]  mail.min.js:148:14867
[navigation] "folder" Array [ 3, "SU5CT1g=", false ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "title:update" Array [  ]  mail.min.js:148:14867
[ui] "messagesview:message:setactive" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[notification] "favicon:change" Array [ "/apps/mail/img/favicon.png" ]  mail.min.js:148:14867
[keyboard] "keyup" Array [ Object, 73 ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[ui] "document:click" Array [ Object ]  mail.min.js:148:14867
[keyboard] "keyup" Array [ Object, 65 ]  mail.min.js:148:14867
[keyboard] "keyup" Array [ Object, 17 ]  mail.min.js:148:14867

	```
</details>
